security/security-secrets-setup.1.rst

```diff
@@ -0,0 +1,191 @@
+
+* The private key of the certificate authority (CA) is shredded (securely erased) prior to caching or deployment to block issuance of new CA descendants (this is most relevant in caching mode).
+
+
+OPTIONS
+=======
+
+Modes of operation
+------------------
+
+  \-config *params.json*
+    Legacy option from prior version of security-secrets-setup.
+    Deploys PKI using existing behavior in Edinburgh release.
+
+  \-generate
+    Causes a PKI to be generated afresh every time and deployed.
+    Typically, this will be whenever the framework is started.
+
+  \-cache
+    Causes a PKI to be generated exactly once and then copied
+    to a designated cache location for future use.
+    The PKI is then deployed from the cached location.
+
+  \-import
+    This option is similar to ``-cache`` in that it deploys a PKI
+    from *CacheDir* to *DeployDir*, but it forces an error if
+    *CacheDir* is empty instead of triggering PKI generation.
+    This enables usage models for deploying a pre-populated PKI
+    such as a Kong certificate signed by an external certificate authority
+    or TLS keys signed by an offline enterprise certificate authority.
+
+  \-h
+    Display help text
+
+  \-help
+    Display help text
+
+  \-confdir confdir
+    Look in this directory for configuration.toml instead.
+
+
+FILES
+=====
+
+edgex-vault.json, edgex-kong.json
+---------------------------------
+
+(Used in generate, cache, and import modes.)
+Configuration files for certificate parameters.
+The basename of this file creates a corresponding directory under *DeployDir*.
+For example, ``edgex-vault.json`` would create assets under
+``/run/edgex/secrets/pki/edgex-vault/``.
+This file conforms to the following schema:
+
+::
+
+    {
+        "create_new_rootca": "true|false",
+        "working_dir": "./config",
+        "pki_setup_dir": "pki",
+        "dump_config": "true",
+        "key_scheme": {
+            "dump_keys": "false",
+            "rsa": "false",
+            "rsa_key_size": "4096",
+            "ec": "true",
+            "ec_curve": "384"
+        },
+        "x509_root_ca_parameters": {
+            "ca_name": "EdgeXFoundryCA",
+            "ca_c": "US",
+            "ca_st": "CA",
+            "ca_l": "San Francisco",
+            "ca_o": "EdgeXFoundry"
+        },
+        "x509_tls_server_parameters": {
+            "tls_host": "edgex-vault|edgex-kong",
+            "tls_domain": "local",
+            "tls_c": "US",
+            "tls_st": "CA",
+            "tls_l": "San Francisco",
+            "tls_o": "Kong"
+        }
+    }
+
+In ``-generate`` or ``-cache`` modes,
+the utility hard-codes the names of the configuration files
+and always processes ``edgex-vault.json`` first
+and ``edgex-kong.json`` second.
+This hard-coding is done for now in order to meet the October Fuji deadline.
```

Comment:
I don't think we should include this point in the docs? Maybe just say 

```suggestion
This will be configurable in the future.
```

Reply:
Suggestion accepted.